pkg/runtime/depot.go

```diff
@@ -146,6 +148,10 @@ func (d *depot) Put(id strfmt.UUID) error {
 
 // DeployViaLink will take an artifact from the depot and link it to the target path.
 func (d *depot) DeployViaLink(id strfmt.UUID, relativeSrc, absoluteDest string) error {
+	if !d.supportsHardLinks {
+		return d.DeployViaCopy(id, relativeSrc, absoluteDest)
+	}
```

Comment:
This needs to happen at a higher level. Recall we record how files were deployed in a config file. We'll need to ensure that this config has the correct value also.

Reply:
I don't see where this can happen. The next level up is in setup.go (https://github.com/ActiveState/cli/blob/84273faa27f55bc371a897ef902b9b2c8f9c9edd/pkg/runtime/setup.go#L466-L477), and I'm not sure we want to expose and make setup change its behavior based on whether or not the depot supports hard links -- it feels like an internal implementation issue.

Also, the storage of deployment type in a config file happens in the DeployVia*() methods, so it will have the correct value.

If there is more to do, please advise.

Reply:
ok, we're gonna have to hoist this up to setup.go then. The depots only concern is making it happen, setup.go owns the logic of interpreting envdef and invoking the appropriate deployer. We may want to revisit that given all the recent changes, but probably not for this story, unless addressing it in setup.go becomes difficult or creates other challenges.